packages/react/src/components/select/select.tsx

```diff
@@ -0,0 +1,68 @@
+import { c, classy, m, SelectProps as BaseProps } from '@onfido/castor';
+import { Icon, useField } from '@onfido/castor-react';
+import React, { useState } from 'react';
+import { withRef } from '../../utils';
+
+let idCount = 0;
+
+/**
+ * `Select` uses an `Icon` that requires `Icons` (SVG sprite) to be included in
+ * your app.
+ *
+ * https://github.com/onfido/castor-icons#use-with-plain-code
+ */
+export const Select = withRef(
+  (
+    {
+      id = `castor_select_${++idCount}`,
+      placeholder,
+      defaultValue,
+      value,
+      borderless,
+      invalid,
+      children,
+      className,
+      onChange,
+      ...restProps
+    }: SelectProps,
+    ref: SelectProps['ref']
+  ): JSX.Element => {
+    const [empty, setEmpty] = useState(!!placeholder);
+    const { disabled, touched } = useField();
+
+    return (
+      <div className={classy(c('select'), m({ borderless }))}>
+        <select
+          disabled={disabled} // will be overriden by props if set
+          {...restProps}
+          ref={ref}
+          id={id}
+          defaultValue={
+            defaultValue || value != null ? undefined : placeholder && ''
+          }
+          value={value}
+          className={classy(
+            c('select-native'),
+            m({ empty, borderless, invalid, touched }),
+            className
+          )}
+          onChange={(ev) => {
+            setEmpty(value != null ? !value : !ev.currentTarget.value);
```

Comment:
This might be a problem if there's an option with a falsy value, like `<option value={0}>Zero<option>`.
It will think it's empty and apply placeholder styles, but it's not empty and it shouldn't.

```suggestion
            setEmpty(value != null ? !value : ev.currentTarget.value == null);
```

Reply:
`ev.currentTarget.value` will always be a string, even in case of setting `value={0}` (or any other number).

Reply:
Ah you're right, nice one.

What about this?

```tsx
<Select placeholder="foo">
  <option value="">None</option>
  <option value="a">A</option>
  <option value="b">B</option>
</Select>
```

Is there any way we could prevent `None` to be styled like a placeholder?
Or do we simply require values for every option?

```tsx
  <option value="none">None</option>
```

Reply:
Option of "None" will be styled same as placeholder, yes. However, it will not be disabled.

I guess the styling is "empty", which is correct - an empty value is being selected.

Reply:
Fair enough, I'm fine with that.
Let's document that `value=""` will always be styled like a placeholder/empty though.

Reply:
Done 👍